compat/test/browser/render.test.js

```diff
@@ -242,6 +242,39 @@ describe('compat render', () => {
 		);
 	});
 
+	// Issue #2582
+	it('should not leak class/className normalisation into props', () => {
+		function Foo({ className, class: cl, ...props }) {
```

Comment:
This pulls class out of props so it won't be forwarded on to the `<li>` even if present.

```suggestion
		function Foo({ className, ...props }) {
			const cl = props.class;
```